src/services/worker_task_tracking_repository.js

```diff
@@ -7,25 +7,52 @@ This Source Code Form is subject to the terms of the Mozilla Public License, v.
 This Source Code Form is “Incompatible With Secondary Licenses”, as defined by the Mozilla Public License, v. 2.0.
 */
 
-
 export default class WorkerTaskTrackingRepository {
   constructor(db) {
     this.db = db;
   }
 
-  async tryToBeginWork(workType) {
-    try {
-      const {insertedId} = await this.db.collection('workerTasks').insertOne({workType});
-      return insertedId;
-    } catch (error) {
-      if (error.message.includes(`duplicate key error`)) {
-        throw new Error('Work of this type is currently in progress');
-      }
-      throw new Error(error);
+  statuses = {
+    running: 'RUNNING',
+    timeout: 'TIMEOUT',
+    success: 'SUCCESS',
+    failure: 'FAILURE'
+  };
+
+  async markExpiredTasks() {
+    await this.db.collection('workerTasks').updateMany({
+      status: this.statuses.running,
+      taskTimeout: {$lte: new Date()}
+    }, {
+      $set: {status: this.statuses.timeout}
+    });
+  }
+
+  async isTaskRunning(workType) {
+    return (await this.db.collection('workerTasks').findOne({
+      status: this.statuses.running,
+      workType
+    })) !== null;
+  }
+
+  async tryToBeginWork(workType, taskTimeout) {
```

Comment:
Previous implementation had one mongo call, this one has 3. This might be a problem if we expect a lot of worker tasks to be started.